### ISSUE TITLE
Accessing posterior chains with attach.jags function

### ISSUE THREAD

**Issue by groemer** [brianstock/MixSIAR#80]
Hi Brian,

I ran the following model after emulating the wolf vignette:

model_filename <- "MixSIAR_model.txt"
> resid_err <- TRUE
> process_err <- TRUE
> write_JAGS_model(model_filename, resid_err, process_err, Laguna.mix, Laguna.source)
> jags.Laguna.verylong <- run_model(run="very long", Laguna.mix, Laguna.source, Laguna.discr, model_filename, 
+                     alpha.prior=c(13,0.01,25.5,11), resid_err, process_err)

But then when I tried to attach the object with attach.jags to get access to the posterior chains, I received the following:

> attach.jags(jags.Laguna.verylong)
Error: could not find function "attach.jags"

Also, if I do not have any factors in the run, would the variable names be different than "p.fac1", p.fac2", etc.

Thanks for helping.

Best,

Gary

**Comment by brianstock** [comment]
Hi Gary,

Without any factors you'll want `p.global`, which should be a matrix with n.draws rows and n.sources columns.  There are 3 ways to get at the MCMC chains.

The `attach.jags` function is in the `R2jags` package.  Load it via:
```
library(R2jags)
attach.jags(jags.Laguna.verylong)
```

Specify the package the function lives in:
```
R2jags::attach.jags(jags.Laguna.verylong)
```

You can also directly access this matrix via:
```
jags.Laguna.verylong$BUGSoutput$sims.list$p.global
```

**Comment by groemer** [comment]
Thanks Brian! 

Clearly I have a long way to go.

Best,

Gary

**Comment by atamkin** [comment]
I've been using the third method to extract my chains: jags.1$BUGSoutput$sims.list$p.fac1
But I have noticed that I only have p.fac[n] when the factor(s) are random. Are the chains not sorted out the same way for fixed variables?
In jags.1$BUGSoutput, I see that there are other objects, like ilr.fac[n], but my searches haven't uncovered what those are.
Thanks!

**Comment by brianstock** [comment]
Hi --

Data files and script: https://github.com/brianstock/MixSIAR/issues/100
With 1 fixed + 1 random effect, the proper proportions to use are created by combining the global intercept, the factor1 offset, and the factor2 offset:

1. `ilr.both[,f1,f2,src] = ilr.global[,src] + ilr.fac1[,f1,src] + ilr.fac2[,f2,src]`
2. Convert from ILR-space to proportion-space (messy)

I was already calculating these in the `output_JAGS` function to create the summary statistics (correctly).  When looking at this, I realized there was a bug in the posterior plotting section though - the plots were getting saved over themselves so you were only left with the last level of factor1. This is fixed now... thanks for uncovering another bug! https://github.com/brianstock/MixSIAR/issues/101

While I was at it, I made the `output_JAGS` function return `p.both`, which is what you want -- the MCMC chains for each factor-level combination. Now if you save the result of `output_JAGS`:
```
p.both <- output_JAGS(jags.1=RG.out,
            mix=data.mix, 
            source=data.source, 
            output_options=output_options)
```
This has dimensions = [n.draws, f1.levels, f2.levels, n.sources]. Check the order of the f1 levels, f2 levels, and sources (alphabetical) using:
```
data.mix$FAC[[1]]$labels
data.mix$FAC[[2]]$labels
data.source$source_names
```
So to get the proportion for Spring - Outflow_30 - SOM:
```
hist(p.both[,2,3,4])
```
Check that this matches the nicer looking posterior plot.
